packages/teams-js/src/public/meeting.ts

```diff
@@ -925,6 +925,124 @@ export namespace meeting {
     );
   }
 
+  /**
+   * This function is used to join a meeting.
+   * This opens a meeting in a new window for the desktop app.
+   * In case of a web app, it will close the current app and open the meeting in the same tab.
+   * @param joinMeetingParams The parameters for joining the meeting. {@link JoinMeetingParams}
+   * @throws error if the meeting join fails, the promise will reject to an object with the error message.
+   */
+
+  export function joinMeeting(joinMeetingParams: JoinMeetingParams): Promise<void> {
+    if (!validateJoinMeetingParams(joinMeetingParams)) {
+      return Promise.reject('Invalid joinMeetingParams');
+    }
+    const { threadId, messageId, replyChainMessageId } = joinMeetingParams.chatInfo;
+
+    joinMeetingParams.joinWebUrl = new URL(joinMeetingParams.joinWebUrl);
+    const serializedJoinMeetingParams = {
+      joinWebUrl: joinMeetingParams.joinWebUrl.href,
+      subject: joinMeetingParams.subject,
+      source: joinMeetingParams.source || EventActionSource.Other,
+      chatInfo: {
+        threadId,
+        messageId: messageId || '0',
+        replyChainMessageId: replyChainMessageId || '0',
+      },
+    };
+    return new Promise<void>((resolve) => {
+      ensureInitialized(runtime);
+      resolve(
+        sendAndHandleStatusAndReason(
+          getApiVersionTag(ApiVersionNumber.V_2, ApiName.Meeting_JoinMeeting),
+          'meeting.joinMeeting',
+          serializedJoinMeetingParams,
+        ),
+      );
+    });
+  }
+
+  /**
+   * This function is used to check the validity of joinMeetingParams which return true if the joinMeetingParams' joinWebUrl and chatInfo are valid else false.
+   * @param joinMeetingParams The parameters for joining the meeting. {@link JoinMeetingParams}
+   * @returns false if joinMeetingParams' joinWebUrl or chatInfo is not valid else true.
+   */
+  function validateJoinMeetingParams(joinMeetingParams: JoinMeetingParams): boolean {
+    if (
+      !joinMeetingParams?.chatInfo?.threadId ||
+      !joinMeetingParams?.joinWebUrl ||
+      joinMeetingParams.chatInfo.threadId === ''
+    ) {
+      return false;
+    }
+    if (typeof joinMeetingParams.joinWebUrl === 'string' && joinMeetingParams.joinWebUrl === '') {
+      return false;
+    }
+
+    if (joinMeetingParams.joinWebUrl instanceof URL && joinMeetingParams.joinWebUrl.href === '') {
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Contains information associated with parameters required for joining the Microsoft Teams meetings.
+   * More details can be found at:
+   * [Online Meeting Base - Microsoft Graph v1.0](https://learn.microsoft.com/en-us/graph/api/resources/onlinemeetingbase?view=graph-rest-1.0)
+   */
+  export interface JoinMeetingParams {
```

Comment:
Based on what we've learned about `subject` and the comment I left in `ChatInfo`, here's what I would suggest for a new structure for this interface:

```typescript
export interface JoinMeetingParams {
  /** The join URL of the online meeting. */
  joinWebUrl: URL | string;
  /** The unique identifier for a thread in Microsoft Teams. Should be the thread id of the chat associated with the meeting being joined*/
  threadId: string;
  /** The source of the join button click. {@link EventActionSource} */
  source: EventActionSource;
  /** The subject to show on the pre-join screen of the meeting. If this value is omitted the meeting's actual subject will be displayed after a brief delay. */
  customPreJoinSubject?: string;
}
```

I've removed the ChatInfo structure entirely because I assume threadId is the only meaningful parameter a developer will specify from it. I've also changed the name of `subject` and added a more detailed comment explaining what it's for and what happens if it isn't specified. I've also made `source` a required parameter (you listed it as optional). If it is *truly* optional, then the comment should explain what happens if the source is not specified.

Reply:
@TrevorJoelHarris,
1. In chatInfo, the replyChainMessageId and messageId are used for the channel meetings and hence are important for joining/opening chat for channel meetings.
2. Will use your suggestion for replacing subject with customPreJoinSubject.
3. We had made source as an optional parameter and if nothing was passed in we assume it to be 'other'. Hence there's no need for source to be optional.

Reply:
In response to 2, you can definitely change the name of subject. There's no technical reason that the names have to match, and in this case the subject has important extra context. The developer doesn't have to pass in the actual value of the meeting's subject, they can pass in anything they want. Right?

Reply:
In response to 3: okay, that's your team's call. If you want to leave it optional with a default of other you can, just keep in mind that once you check this in and people start using it you can't change your mind and make the parameter required later.

Reply:
Only open issue is 2, going to resolve this and open a new item tracking that